src/domain/state.ts

```diff
@@ -901,6 +901,18 @@ function updateAppStateForReviewApplication(
   }
 }
 
+export function signatureOnlySectionRequiringRevisions(application: Application) {
+  const sectionsWithRevisions = Object.keys(application.revisionRequest)
+    .filter((sec) => sec !== 'general')
+    .filter(
+      (section) => application.revisionRequest[section as keyof RevisionRequestUpdate].requested,
```

Comment:
There is a chance for an error here, if `revisisonRequest[section]` is undefined... you are casting the `section` into a keyof the type you are interested in, but can we be certain that will be the case?

If you can convince me this is safe can we not also convince TS?